orchagent/routeorch.cpp

```diff
@@ -2547,3 +2565,968 @@ void RouteOrch::decNhgRefCount(const std::string &nhg_index)
         gCbfNhgOrch->decNhgRefCount(nhg_index);
     }
 }
+
+void RouteOrch::generateRouteFlowStats()
+{
+    SWSS_LOG_ENTER();
+    for (const auto &route_pattern : mRoutePatternSet)
+    {
+        createRouteFlowCounterByPattern(route_pattern, 0);
+    }
+}
+
+void RouteOrch::clearRouteFlowStats()
+{
+    SWSS_LOG_ENTER();
+    if (!mBoundRouteCounters.empty() || !mPendingAddToFlexCntr.empty())
+    {
+        std::list<RouteFlowCounterBulkContext> to_unbind;
+        for (auto &entry : mBoundRouteCounters)
+        {
+            const auto& route_pattern = entry.first;
+            for (auto &inner_entry : entry.second)
+            {
+                removeRouteFlowCounterFromDB(route_pattern.vrf_id, inner_entry.first, inner_entry.second);
+                bulkUnbindFlowCounters(route_pattern.vrf_id, inner_entry.second, inner_entry.first, to_unbind);
+            }
+        }
+
+        for (auto &entry : mPendingAddToFlexCntr)
+        {
+            const auto& route_pattern = entry.first;
+            for (auto &inner_entry : entry.second)
+            {
+                bulkUnbindFlowCounters(route_pattern.vrf_id, inner_entry.second, inner_entry.first, to_unbind);
+            }
+        }
+
+        flushUnbindFlowCounters(to_unbind);
+
+        mBoundRouteCounters.clear();
+        mPendingAddToFlexCntr.clear();
+    }
+
+    mUnboundRoutes.clear();
+}
+
+void RouteOrch::updateRoutePattern(const std::string &pattern, size_t max_match_count)
```

Comment:
The updateRoutePattern seems to add new pattern to the list, rather than replacing existing pattern. The requirement of 1 pattern per IP type is handled at CLI and so the backend is not enforcing it leaving for easy migration to multiple patterns per type in the future. Am I correct here?
If that's the case can we rename the API to addRoutePattern?